omegaconf/basecontainer.py

```diff
@@ -245,7 +268,7 @@ def is_float(data: str) -> bool:
             return dumper.represent_scalar(
                 yaml.resolver.BaseResolver.DEFAULT_SCALAR_TAG,
                 data,
-                style=("single-quoted" if with_quotes else None),
+                style=("'" if with_quotes else None),
```

Comment:
What is with_quotes? where is it coming from?

Reply:
Since bool, float and int are the ones causing the problem we are quoting them if they are strings. If a string is not one of them its not necessary to quote.

Reply:
I understand, my question is about the usage of the variable `with_quotes`. where is it defined?

Reply:
Ah, I missed the definition (it was folded).